WordPress/Classes/Services/ReaderPostService.m

```diff
@@ -324,15 +324,15 @@ - (void)setFollowing:(BOOL)following
 }
 
 - (void)toggleFollowingForPost:(ReaderPost *)post
-                       success:(void (^)(void))success
-                       failure:(void (^)(NSError *error))failure
+                       success:(void (^)(BOOL follow))success
+                       failure:(void (^)(BOOL follow, NSError *error))failure
 {
     // Get a the post in our own context
     NSError *error;
     ReaderPost *readerPost = (ReaderPost *)[self.managedObjectContext existingObjectWithID:post.objectID error:&error];
     if (error) {
         if (failure) {
-            failure(error);
+            failure(true, error);
```

Comment:
Should this be `follow`?  If I'm following the logic correctly, by setting it to `true` it looks like `dispatchToggleFollowSiteMessage` would only ever dispatch `NoticeMessages.followFail`.  Am I missing something?

Reply:
This is something that I had particular difficulty in "resolving", if we can even consider it resolved.  Please be critical about what I say below, since I may be mistaken.

### First some context about what I did here, and why:

My understanding is that I can't know the follow value I should be returning here, because the `ReaderPost` object no longer exists.

With this in mind, and considering this is an odd scenario and (I believe) such an edge case, and in lack of a simple and good solution, I opted for returning `true` here - with the risk of showing the wrong error message to the user.  If we decide to move forward with this, I can add a comment to explain.

I think the ideal solution would be to ditch the `toggle...` methods in favour of more deterministic `follow...` and `unfollow...` methods, since I think that'd offer us a much easier mechanism to know the user's intention - which I clear in the beginning but becomes shady because of the `toggle...` logic.

The only problem with this approach is that it would not be a minor change.

### A question for you:

> Should this be `follow`?

Could I ask you to expand on what you mean by this precisely?  Maybe even show it to me in terms of code?

I agree with what you're saying, but the follow value isn't known here (unless maybe we pass it as an input parameter, kind of defeating the purpose of the `toggle...` method name).

Reply:
> Could I ask you to expand on what you mean by this precisely? 

Ugh. This was me mis-reading the method signature and interpreting `follow` as a method argumement, rather than a block argument. 🤦   Mea Culpa. Sorry for the confusion there. 

> First some context about what I did here

Thanks for that :).  It makes perfect sense to me now.  I agree that getting away from the toggle approach would be best, and I agree that it's a larger change that's out of scope. :)  
Let's ship this as it does resolve the issue and, as you pointed out, it's unlikely for a user to encounter a missing model object. (And if they do we probably have a bigger problem.)

Reply:
Thanks Eric!  I appreciate the time and guidance!